docs/how_to/documenting_new_feature_toggles.rst

```diff
@@ -0,0 +1,94 @@
+
+- Generation of human-readable documentation of existing feature toggles for Open edX operators: instead of grepping a large codebase, operators (sysadmins and product owners, for instance) should be able to search an online document to learn how to use and enable features on their platforms.
+- Issue detection and resolution at runtime: feature toggles are exposed via an API on a running platform. This allows for automated discovery of feature toggles and, for instance, comparing the state of feature toggles between different environments at runtime. Annotations will allow operators to quickly understand the nature of the discrepancies, and to address issues more efficiently.
+
+Examples
+========
+
+Boilerplate template
+--------------------
+
+Copy-paste this boilerplate template to document a feature toggle::
+
+    # .. toggle_name: SOME_FEATURE_NAME
+    # .. toggle_implementation: WaffleFlag OR WaffleSwitch OR CourseWaffleFlag OR ExperimentWaffleFlag OR ConfigurationModel OR DjangoSetting
+    # .. toggle_default: True OR False
+    # .. toggle_description: Add here a detailed description of the consequences of enabling this feature toggle.
+    #   Note that all annotations can be spread over multiple lines by prefixing every line after the first by
+    #   at least two spaces, plus the leading space.
+    # .. toggle_warnings: Add here additional instructions that users should be aware of. For instance, dependency
+    #   on additional settings or feature toggles should be referenced here. If this field is not required, simply remove it.
+    # .. toggle_use_cases: temporary OR circuit_breaker OR vip OR opt_out OR opt_in OR open_edx
+    # .. toggle_creation_date: 2020-12-31
+    # .. toggle_target_removal_date: 2021-06-31 (this is required if toggle_use_cases is temporary. If not, simply remove it.)
+    # .. toggle_tickets: https://openedx.atlassian.net/browse/TICKET-xxx OR https://github.com/edx/edx-platform/pull/xxx
+    SOME_FEATURE_NAME = ...
+
+Configuration model
+-------------------
+
+Annotations can also be written in docstrings. This is particularly useful for documenting a ConfigurationModel class, for instance::
+
+    class SomeFeatureFlag(ConfigurationModel):
+        """
+        .. toggle_name: SomeFeatureFlag.enabled
+        .. toggle_implementation: ConfigurationModel
+        ...
+        """
+
+Annotation fields
+=================
+
+Most annotation fields are self-explanatory. In this section we describe in more details some of the more complex annotation fields.
+
+``toggle_implementation``
+-------------------------
+
+Must be one of the following:
+
+- "WaffleFlag", "WaffleSwitch", "CourseWaffleFlag": for objects that inherit from ``waffle_utils``' ``WaffleFlag``, ``WaffleSwitch`` or ``CourseWaffleFlag``, respectively.
```

Comment:
Can waffle_utils be a link to the directory or a relevant doc?

Reply:
:ok_hand: